lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -94,6 +96,28 @@ bool PreprocessMetadata::runOnModule(Module &Module) {
   return true;
 }
 
+void PreprocessMetadata::preprocessCXXStructorList(
+    SPIRVMDBuilder::NamedMDWrapper &EM, GlobalVariable *V,
+    ExecutionMode EMode) {
+  auto *List = dyn_cast_or_null<ConstantArray>(V->getInitializer());
+  if (!List)
+    return;
+
+  for (Value *V : List->operands()) {
+    auto *Structor = dyn_cast<ConstantStruct>(V);
+    if (!Structor)
+      continue;
```

Comment:
Two questions:
- do we really need to check each operand? `ConstantArray` holds objects of the same time
- do we even need a runtime check? Would an assertion be enough?

Reply:
Perhaps a non-dynamic `cast` would be sufficient indeed (which will include the assert in itself).